pkg/controller/controller.go

```diff
@@ -33,8 +33,12 @@ const (
 
 // MachineApproverReconciler reconciles a machine-approver  object
 type CertificateApprover struct {
-	client.Client
-	RestCfg  *rest.Config
+	NodeClient  client.Client
+	NodeRestCfg *rest.Config
```

Comment:
I know the NodeRestCfg is used by the `approve()` signature but seems unnecessary we we already have the clients here, so we can reuse the nodeClient or build a csrClient.